### ISSUE TITLE
a11y: Fix accessibility issues identified in a11y report from a11y test suite

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#17279]
Accessibility test suite (e2e) runs on PRs and master branch.

It identifies a11y issues - we need to address those.

Current status:

## Accessibility Violations Summary

| Severity | Count |
|----------|-------|
| 🔴 Critical | 65 |
| 🟠 Serious | 123 |
| 🔵 Moderate | 21 |
| ⚪ Minor | 13 |
| **Total** | **222** |

## Violations Detail

| Level | Violation | Count | Description |
|-------|-----------|-------|-------------|
| 🟠 Serious | `color-contrast` | 2 | Ensure the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds |
| 🔵 Moderate | `heading-order` | 17 | Ensure the order of headings is semantically correct |
| 🔴 Critical | `aria-required-parent` | 51 | Ensure elements with an ARIA role that require parent roles are contained by them |
| 🔴 Critical | `aria-valid-attr-value` | 2 | Ensure all ARIA attributes have valid values |
| 🟠 Serious | `list` | 1 | Ensure that lists are structured correctly |
| 🟠 Serious | `nested-interactive` | 79 | Ensure interactive controls are not nested as they are not always announced by screen readers or can cause focus problems for assistive technologies |
| ⚪ Minor | `empty-table-header` | 7 | Ensure table headers have discernible text |
| 🟠 Serious | `aria-dialog-name` | 7 | Ensure every ARIA dialog and alertdialog node has an accessible name |
| 🔴 Critical | `aria-required-children` | 11 | Ensure elements with an ARIA role that require child roles contain them |
| 🟠 Serious | `listitem` | 34 | Ensure &lt;li&gt; elements are used semantically |
| 🔵 Moderate | `landmark-unique` | 2 | Ensure landmarks are unique |
| ⚪ Minor | `image-redundant-alt` | 1 | Ensure image alternative is not repeated as text |
| 🔵 Moderate | `landmark-main-is-top-level` | 1 | Ensure the main landmark is at top level |
| 🔵 Moderate | `landmark-no-duplicate-main` | 1 | Ensure the document has at most one main landmark |
| ⚪ Minor | `empty-heading` | 5 | Ensure headings have discernible text |
| 🔴 Critical | `aria-allowed-attr` | 1 | Ensure an element's role supports its ARIA attributes |
